File: content/atlas/source/includes/fts/field-types/embeddedDocuments/stored-source-mltpl-conf.java
import com.mongodb.client.*;
import org.bson.Document;
import java.util.Arrays;

public class CreateIndex {
    public static void main(String[] args) {
        // Replace the connection string with your MongoDB deployment's connection string
        String connectionString = "<connection-string>";

        try (MongoClient mongoClient = MongoClients.create(connectionString)) {
            MongoDatabase database = mongoClient.getDatabase("sample_training");
            MongoCollection<Document> collection = database.getCollection("companies");
            String indexName = "default";

            // Create the Atlas Search index definition for the embeddedDocuments field
            Document searchIdx = new Document(
                "mappings",
                new Document("dynamic", false)
                    .append("fields",
                        new Document("products",
                            new Document("type", "embeddedDocuments")
                                .append("dynamic", true)
                                .append("storedSource", true)
                        )
                    )
            ).append("storedSource",
                new Document("include", Arrays.asList("_id", "name"))
            );
            collection.createSearchIndex(indexName, searchIdx);
            System.out.println("New index name: " + indexName);
        } catch (Exception e) {
            System.err.println("Error creating Atlas Search index: " + e.getMessage());
        }
    }
}

File: content/search/source/includes/index/field-types/embeddedDocuments/code-snippets/python/stored-source-mltpl-conf.py
from pymongo import MongoClient
from pymongo.operations import SearchIndexModel

# Connect to your Atlas deployment
uri = "<connection-string>"
client = MongoClient(uri)

database = client["sample_training"]
collection = database["companies"]

# Create the Atlas Search index definition for the embeddedDocuments field
search_index_model = SearchIndexModel(
    definition={
        "mappings": { 
          "dynamic": false,
          "fields": { 
            "products": { 
              "type": "embeddedDocuments",
              "dynamic": true,
              "storedSource": true
            } 
          }
        },
        "storedSource": {
          "include": [ "_id", "name" ]
        }
    },
    name="default",
)

# Create the index
result = collection.create_search_index(model=search_index_model)
print(f"New index name: {result}")
